type-generator/src/main/java/io/mesosphere/types/dtr/compiler/internal/CompiledFragment.java

```diff
@@ -0,0 +1,29 @@
+package io.mesosphere.types.dtr.compiler.internal;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * A compiled fragment is instantiated from the compiler engine and can be
+ * evaluated at any time.
+ */
+public interface CompiledFragment {
```

Comment:
Could you give an example?

Reply:
Initially, I thought of creating something that can be evaluated in a getter, allowing templates to render other templates. Therefore you could pass a "CompiledFragment" to a template argument, and render it when needed.

But I then realised that something like this can be easily achieved with StringTemplate if given values with enough information (and that was the beginning of the `*Arguments` classes).

Therefore I agree that this could be replaced with something like `engine.render(template, arguments)`